### PR TITLE
Added Mark as Completed button

### PR DESCRIPTION
**Pull request by MaharshM**

### CODE REVIEW

File: app/Http/Requests/SaveTodoRequest.php [hunk 1]
```diff
@@ -50,6 +50,11 @@ public function rules()
                 'date',
             ],
 
+            'completed_at' => [
```
Comment (abbasali) [inline app/Http/Requests/SaveTodoRequest.php:53]: Why do we need this? I believe it is not used anywhere.

File: routes/web.php [hunk 2]
```diff
@@ -30,3 +30,4 @@
 Route::resource('category.todo', TodoController::class);
 Route::get('/attachment/{attachment}/download', [AttachmentController::class, 'download'])->name('attachment.download');
 Route::resource('attachment', AttachmentController::class)->only(['destroy']);
+Route::put('/category/{category}/todo/{todo}', [TodoController::class, 'markAsCompleted'])->name('category.todo.markAsCompleted');
```
Comment (abbasali) [inline routes/web.php:33]: The URL for this route should be `PUT /todo/{todo}/complete`.

File: app/Http/Controllers/TodoController.php [hunk 3]
```diff
@@ -142,4 +145,14 @@ public function destroy(Todo $todo)
     {
         //
     }
+
+    public function markAsCompleted(Request $request, Category $category, Todo $todo)
+    {
```
Comment (abbasali) [inline app/Http/Controllers/TodoController.php:150]: As always... where is the authorization? Can any user mark any other user's todo as completed?

**Review by abbasali — CHANGES_REQUESTED** [review]

File: routes/web.php (hunk 2 above)
Comment (MaharshM) [reply routes/web.php:33]: done.

File: app/Http/Controllers/TodoController.php (hunk 3 above)
Comment (MaharshM) [reply app/Http/Controllers/TodoController.php:150]: done.

File: app/Http/Requests/SaveTodoRequest.php (hunk 1 above)
Comment (MaharshM) [reply app/Http/Requests/SaveTodoRequest.php:53]: done.

**Review by abbasali — APPROVED** [review]

**Pull request merged**